src/BundleAnalyzerPlugin.js

```diff
@@ -134,16 +134,23 @@ class BundleAnalyzerPlugin {
   }
 
   getBundleDirFromCompiler() {
-    switch (this.compiler.outputFileSystem.constructor.name) {
-      case 'MemoryFileSystem':
-        return null;
-      // Detect AsyncMFS used by Nuxt 2.5 that replaces webpack's MFS during development
-      // Related: #274
-      case 'AsyncMFS':
-        return null;
-      default:
-        return this.compiler.outputPath;
+    // On Webpack 4.x, compiler.outputFileSystem is a class instance
+    if (this.compiler.outputFileSystem.constructor) {
+      switch (this.compiler.outputFileSystem.constructor.name) {
+        // Detect memory-fs
+        // var MemoryFileSystem = require("memory-fs")
+        // compiler.outputFileSystem = new MemoryFileSystem()
+        case 'MemoryFileSystem':
+          return null;
+        // Detect AsyncMFS used by Nuxt 2.5 that replaces webpack's MFS during development
+        // Related: #274
+        case 'AsyncMFS':
+          return null;
+      }
     }
+    // TODO Can we check the outputFileSystem is graceful-fs or memfs on webpack 5.x?
+    // If avaliable, we can return null if the outputFileSystem is memfs.
+    return this.compiler.outputPath;
```

Comment:
The thing is in webpack 5 it will return `this.compiler.outputPath` even in dev mode (when `webpack-dev-server` is used) and will try to parse compiled bundle which is not there.
Seems like we need some other way to detect non-physical file systems now.